translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+
+        if (mqttMessage.getPayload().getBody().length == 0) {
+            assertTrue(mqttMessage.getPayload().getBody().length == 0);
+        } else {
+            assertTrue(mqttMessage.getPayload().getBody().length != 0);
+        }
+    }
+
+    @And("^I got kura response message channel with \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\" and \"(.+)\" data$")
+    public void iGotKuraResponseMessageChannelWithAndData(String replyPart, String requestId, String appId, String messageClassification, String scope, String clientId) {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getChannel().getReplyPart().equals(replyPart));
+        assertTrue(kuraResponseMessage.getChannel().getRequestId().equals(requestId));
+        assertTrue(kuraResponseMessage.getChannel().getAppId().equals(appId));
+        assertTrue(kuraResponseMessage.getChannel().getMessageClassification().equals(messageClassification));
+        assertTrue(kuraResponseMessage.getChannel().getScope().equals(scope));
+        assertTrue(kuraResponseMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and with metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithoutBodyAndWithMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
+
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload with body and metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithBodyAndMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.setBody("Payload".getBytes());
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
+
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @Given("^I try to translate mqtt message to kura data message$")
+    public void iTryToTranslateMqttMessageToKuraMessage() throws Exception {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+        KuraDataMessage kuraDataMessage = translatorDataMqttKura.translate(mqttMessage);
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.